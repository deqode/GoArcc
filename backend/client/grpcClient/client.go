package grpcClient

import (
	"alfred/config"
	"alfred/logger"
	"alfred/servers/openTracing/tracer/jaeger"
	"context"
	grpc_opentracing "github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing"
	"go.uber.org/zap"
	"google.golang.org/grpc"
	"time"
)

//ClientContext to store context
type ClientContext struct {
	Ctx context.Context
}

//GetGrpcClientConnection is used to get the client connection
func GetGrpcClientConnection(config *config.Config) *grpc.ClientConn {
	var opts []grpc.DialOption
	opts = append(opts,
		grpc.WithUnaryInterceptor(
			//tracing for unary interceptors
			grpc_opentracing.UnaryClientInterceptor(grpc_opentracing.WithTracer(jaeger.Tracer)),
		),
		//tracing for stream interceptors
		grpc.WithStreamInterceptor(grpc_opentracing.StreamClientInterceptor(grpc_opentracing.WithTracer(jaeger.Tracer))),
	)

	//append grpc insecure
	opts = append(opts,
		grpc.WithInsecure(),
	)
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(config.Grpc.RequestTimeout))
	defer cancel()

	// You must have some sort of OpenTracing Tracer instance on hand.
	conn, err := grpc.DialContext(ctx, config.Grpc.Host+":"+config.Grpc.Port, opts...)
	if err != nil {
		logger.Log.Fatal("did not connect", zap.Error(err))
	}
	return conn
}
