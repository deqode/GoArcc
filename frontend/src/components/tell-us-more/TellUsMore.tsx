import '../../css/bootstrap.min.css'
import '../../css/alfred.css';
import '../../css/fonts.css';
import logo from '../../images/logo.png';

function TellUsMore() {
  return (
    <div>
      <nav>
        <div className="container-fluid">
          <div className="logo">
            <a href="/" className="d-block"><img src={logo} alt="Logo!" /></a>
          </div>
        </div>
      </nav>

      <section className="content_section">
        <div className="container">
          <div className="row">
            <div className="col-md-12 project-info_form">
              <h1 className="page-head">Tell Alfed more about your project.</h1>
              <form className="form container">
                <div className="form-group row custom_input custom_input">
                  <label htmlFor="appname" className="col-md-4">Your App's Name</label>
                  <input type="text" className="form-control col-md-8" placeholder="Demo App" />
                </div>
                <div className="form-group row select custom_input">
                  <label className="col-md-4" htmlFor="inlineFormCustomSelectPref">Your Repo's Name</label>
                  <select className="custom-select col-md-8" id="inlineFormCustomSelectPref" defaultValue="default">
                    <option value="default">email finder</option>
                    <option value="1">One</option>
                    <option value="2">Two</option>
                    <option value="3">Three</option>
                  </select>
                </div>
                <div className="form-group row select custom_input">
                  <label className="col-md-4" htmlFor="inlineFormCustomSelectPref">Your Environment</label>
                  <select className="custom-select col-md-8" id="inlineFormCustomSelectPref" defaultValue="default">
                    <option value="default">Production</option>
                    <option value="1">One</option>
                    <option value="2">Two</option>
                    <option value="3">Three</option>
                  </select>
                </div>
                <div className="form-group row custom_input">
                  <label htmlFor="branchname" className="col-md-4">Your Branch</label>
                  <input type="text" className="form-control col-md-8" placeholder="master" />
                </div>
              </form>
              <div className="text-center mt-5 mb-5"><a href="/success" className="btn fetch-btn w-50">Go Fetch</a></div>
            </div>
          </div>
        </div>
      </section>
    </div>
  )
}

export default TellUsMore
