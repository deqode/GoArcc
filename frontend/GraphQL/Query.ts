import { gql } from '@apollo/client';

export const GET_REPOSITORIES = gql`
query repositories(
    $userid:String!
    $accountid:String!
    $provider:Types_Enum_GitProviders!
    ){
    repositories(
      provider:$provider,
      user_id:$userid,
      account_id:$accountid,
    ){repositories{
      name
    }}
  }
`
// todo : gql file

export const GET_BRANCHES = gql`
query repository(
    $ownerName:String!
    $repoName:String!
    $accountid:String!
    $provider:Types_Enum_GitProviders!
){
    repository(
      owner_name:$ownerName
      provider:$provider,
      repo_name:$repoName
      account_id:$accountid
    ){
      branches
    }
  }`

export const VCS_CONNECTIONS = gql`
query VCSConnections(
  $accountid:String!
  $provider:Types_Enum_GitProviders!
){
  VCSConnections(
    account_id:$accountid
    provider:$provider,
    ){
    vcs_connection{
            user_name
    }
  }
}`