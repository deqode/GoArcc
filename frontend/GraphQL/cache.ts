import { gql, InMemoryCache, makeVar } from '@apollo/client'

export const token = makeVar(null)

export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        token: {
          read() {
            return token()
          },
        },
      },
    },
  },
})

export const GET_TOKEN = gql`
  query getToken {
    token @client
  }
`
