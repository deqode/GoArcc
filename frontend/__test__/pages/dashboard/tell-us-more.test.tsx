import { MockedProvider } from '@apollo/client/testing'
import { act, create, ReactTestRenderer } from 'react-test-renderer'
import { GET_BRANCHES, GET_REPOSITORIES } from '../../../GraphQL/Query'
import TellUsMore from '../../../pages/dashboard/tell-us-more'
const props = {
  user: { accessToken: 'accessToken', idToken: 'idToken', userId: 'userId' },
  accountId: 'accountId',
  ownerName: 'accountId',
}

const mocks = [
  {
    request: {
      query: GET_REPOSITORIES,
      variables: {
        provider: 'provider',
        userId: 'userId',
        accountId: 'accountId',
      },
    },
    result: {
      data: {
        repositories: {
          repositories: ['REPO1', 'REPO2'],
        },
      },
    },
  },
  {
    request: {
      query: GET_BRANCHES,
      variables: {
        provider: 'provider',
        ownerName: 'ownerName',
        repoName: 'repoName,',
        accountId: 'accountId',
      },
    },
    result: {
      data: {
        repository: {
          branches: ['Branch1', 'Branch1'],
        },
        clone_url: 'Mockclone_url',
      },
    },
  },
]

//TODO: check if gql queries are mocked
describe('dashboard/tell-us-more page tests', () => {
  let wrapper: ReactTestRenderer

  beforeEach(() => {
    act(() => {
      wrapper = create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <TellUsMore {...props} />
        </MockedProvider>
      )
    })
  })
  it('should match the snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
