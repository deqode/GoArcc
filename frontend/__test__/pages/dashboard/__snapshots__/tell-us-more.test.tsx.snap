// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard/tell-us-more page tests should match the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <h1
          className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
        >
          Tell Us More
        </h1>
        <br />
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
        >
          You have so many cool projects.Let me know the one you want to deploy today 
        </h5>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <p
          className="MuiTypography-root makeStyles-text-3 MuiTypography-body1 MuiTypography-displayInline"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-start"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
            >
              <div
                className="MuiFormControl-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                  id="demo-simple-select-outlined-label"
                >
                  Repo Name
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-4 Mui-disabled Mui-disabled MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <div
                    aria-disabled="true"
                    aria-haspopup="listbox"
                    aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
                    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled Mui-disabled"
                    id="demo-simple-select-outlined"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={null}
                    role="button"
                    tabIndex={null}
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </div>
                  <input
                    aria-hidden={true}
                    className="MuiSelect-nativeInput"
                    onAnimationStart={[Function]}
                    onChange={[Function]}
                    required={false}
                    tabIndex={-1}
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined Mui-disabled"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-7"
                    >
                      <span>
                        Repo Name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
            >
              <div
                className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Go Fetch
                </span>
              </button>
            </div>
          </div>
        </p>
      </div>
    </div>
  </div>
</div>
`;
